File: library/src/main/java/com/larryhsiao/aura/sercurity/IsDeviceSecure.java
package com.larryhsiao.aura.sercurity;

import android.app.KeyguardManager;
import android.content.Context;
import android.os.Build;
import com.larryhsiao.clotho.Source;

import static androidx.core.content.ContextCompat.getSystemService;

public class IsDeviceSecure implements Source<Boolean> {
    private final Context context;

    public IsDeviceSecure(Context context) {
        this.context = context;
    }

    @Override
    public Boolean value() {
        final KeyguardManager manager = getSystemService(context, KeyguardManager.class);
        if (manager != null) {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
                return manager.isDeviceSecure();
            } else {
                return manager.isKeyguardSecure();
            }
        } else {
            return false;
        }
    }
}

File: library/src/main/java/com/silverhetch/aura/sercurity/IsDeviceSecure.kt
package com.silverhetch.aura.sercurity

import android.app.KeyguardManager
import android.content.Context
import android.os.Build
import androidx.core.content.ContextCompat
import com.silverhetch.clotho.Source

/**
 * Source to determine if the device have pin/password
 */
class IsDeviceSecure(
    private val context: Context
) : Source<Boolean> {
    override fun value(): Boolean {
        val keyguardManager = ContextCompat.getSystemService<KeyguardManager>(
            context,
            KeyguardManager::class.java
        ) ?: return false
        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            keyguardManager.isDeviceSecure
        } else {
            keyguardManager.isKeyguardSecure
        }
    }
}